tests/src/test/java/org/eclipse/hono/tests/http/TelemetryHttpIT.java

```diff
@@ -199,6 +201,8 @@ public void testUploadQos1MessageFailsIfDeliveryStateNotUpdated(
             final VertxTestContext ctx)
             throws InterruptedException {
 
+        final AmqpApplicationClient amqpApplicationClient = (AmqpApplicationClient) helper.applicationClient;
```

Comment:
why is this cast necessary?

Reply:
Because the test makes verifications on AMQP message specific conditions (by checking the remotely settled flag and the message delivery). Hence it needs to create an AMQP specific telemetry consumer.

Reply:
Ah, I see. It is further down below in the code that has not been changed. That's why I didn't see it at first in the GitHub diff ...